plugins/pulp_rpm/plugins/importers/iso/importer.py

```diff
@@ -129,10 +127,14 @@ def upload_unit(self, transfer_repo, type_id, unit_key, metadata, file_path, con
         :param config: plugin configuration for the repository
         :type  config: pulp.plugins.config.PluginCallConfiguration
         """
-        if models.ISO.objects.filter(**unit_key).count() > 0:
-            return {'success_flag': False, 'summary': _('ISO already exists'), 'details': None}
+        qs = models.ISO.objects.filter(**unit_key)
+        if qs:
+            # iso with this key already exists, use it
+            iso = qs.first()
```

Comment:
I'm not confident in making those changes without causing another regression elsewhere. I had similar thoughts for improving the logic here when working on this, but opted to only fix the specific regression mentioned in the bug rather than engage in a(nother) refactor. For the changes proposed in this review, I suggest opening a new issue with these suggestions for refactoring this ISO unit upload code.